mongodbatlas/fw_resource_mongodbatlas_database_user.go

```diff
@@ -165,6 +165,7 @@ func (r *DatabaseUserRS) Schema(ctx context.Context, req resource.SchemaRequest,
 					Attributes: map[string]schema.Attribute{
 						"collection_name": schema.StringAttribute{
 							Optional: true,
+							Default:  stringdefault.StaticString(""),
```

Comment:
What do we send to the endpoint when the user does not provide collection_name?

Reply:
I see that the param is not required in the API [endpoint](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Database-Users/operation/createDatabaseUser) so I have a couple of thoughts:

- the collection_name should not be compute
I tested this quickly and the endpoint does not return collection_name when it is not provided in the request
```
{
  "awsIAMType": "NONE",
  "databaseName": "admin",
  "groupId": "63f4d4a47baeac59406dc131",
  "ldapAuthType": "NONE",
  "links": [
    {
      "href": "https://cloud-dev.mongodb.com/api/atlas/v2/groups/63f4d4a47baeac59406dc131/databaseUsers/admin/myUser",
      "rel": "self"
    }
  ],
  "oidcAuthType": "NONE",
  "roles": [
    {
      "databaseName": "admin",
      "roleName": "clusterMonitor"
    },
    {
      "databaseName": "admin",
      "roleName": "backup"
    }
  ],
  "username": "myUser",
  "x509Type": "NONE"
}
```

- We should not set default here because collection_name is not required
- The bug may be in [newMongoDBAtlasRoles](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/0694a78eaeb73783ea2a115917dda88f17549e35/mongodbatlas/fw_resource_mongodbatlas_database_user.go#L477)
```golang
	for i, v := range roles {
		out[i] = matlas.Role{
			RoleName:       v.RoleName.ValueString(),
			DatabaseName:   v.DatabaseName.ValueString(),
			CollectionName: v.CollectionName.ValueString(),
		}
	}
```
Here we are collection name where we may need instead to check for null.